Java3 homework6

darujo opened the pull request:
1. Добавить на серверную сторону чата логирование, с выводом информации о действиях на сервере (запущен, произошла ошибка, клиент подключился, клиент прислал сообщение/команду).
2. Написать метод, которому в качестве аргумента передается не пустой одномерный целочисленный массив. Метод должен вернуть новый массив, который получен путем вытаскивания из исходного массива элементов, идущих после последней четверки. Входной массив должен содержать хотя бы одну четверку, иначе в методе необходимо выбросить RuntimeException. Написать набор тестов для этого метода (по 3-4 варианта входных данных). Вх: [ 1 2 4 4 2 3 4 1 7 ] -> вых: [ 1 7 ].
3. Написать метод, который проверяет состав массива из чисел 1 и 4. Если в нем нет хоть одной четверки или единицы, то метод вернет false; Написать набор тестов для этого метода (по 3-4 варианта входных данных).

Commit:
java3_homework6
1. Добавить на серверную сторону чата логирование, с выводом информации о действиях на сервере (запущен, произошла ошибка, клиент подключился, клиент прислал сообщение/команду).

Commit:
java3_homework6
2. Написать метод, которому в качестве аргумента передается не пустой одномерный целочисленный массив. Метод должен вернуть новый массив, который получен путем вытаскивания из исходного массива элементов, идущих после последней четверки. Входной массив должен содержать хотя бы одну четверку, иначе в методе необходимо выбросить RuntimeException. Написать набор тестов для этого метода (по 3-4 варианта входных данных). Вх: [ 1 2 4 4 2 3 4 1 7 ] -> вых: [ 1 7 ].
3. Написать метод, который проверяет состав массива из чисел 1 и 4. Если в нем нет хоть одной четверки или единицы, то метод вернет false; Написать набор тестов для этого метода (по 3-4 варианта входных данных).

Bakatkin-Alexandr commented on ClientChat/src/main/java/com/darujo/clietnchat/ClientChat.java:
```diff
@@ -30,7 +32,7 @@ public class ClientChat extends Application {
     private FXMLLoader changePasswordWindow;
     private static ClientChat INSTANCE;
-
+    private static final Logger LOGGER = LogManager.getLogger(Network.class);
```
Имя логгера не корректно выбрали - необходимо проставлять класс, в котором вы и будете логировать (в данном случае это ClientChat.class). Потому как это позволяет для каждого логгера устанавливать различные настроки (уровни логирования, форматирование и т.д.)

Bakatkin-Alexandr commented on ClientChat/src/main/java/com/darujo/clietnchat/ClientChat.java:
```diff
@@ -177,7 +179,7 @@ public static ClientChat getInstance() {
 
     public static void showMessage(String message) {
-        System.out.println(message);
+        LOGGER.warn( message);
```
Сначала подумал, что здесь warn() не нужен, потом посмотрел код - мне кажется, у вас не совсем удачное название метода showMessage() - кажется, будто тут отображаются все сообщения.

Bakatkin-Alexandr commented on Network/src/com/darujo/network/ClientHandler.java:
```diff
@@ -17,14 +19,14 @@ public class ClientHandler {
     private final Set<ReaderMessage> listeners;
-
-
+    private static final Logger LOGGER = LogManager.getLogger(Network.class);
```
Тут тоже вопрос с Network.class.

Bakatkin-Alexandr commented on Network/src/com/darujo/network/ClientHandler.java:
```diff
@@ -44,6 +46,7 @@ public void readSocket() {
             try {
                 while (!Thread.currentThread().isInterrupted()) {
                     Command obj = readCommand();
+                    LOGGER.info("Получена команда " + (obj != null ? obj.getType() : ""));
```
Тут можно попробовать использовать Optional, тогда уйдете от NullPointerException. Это не замечание, а просто на подумать.

Bakatkin-Alexandr commented on Network/src/com/darujo/network/ClientHandler.java:
```diff
@@ -59,7 +62,8 @@ public void readSocket() {
                 } catch (IOException e) {
-                    e.printStackTrace();
+                    LOGGER.error(e.getMessage());
+                    LOGGER.error(e.getStackTrace());
```
Почему два логирования? Почему нельзя в одно объединить?

Bakatkin-Alexandr commented on Network/src/com/darujo/network/Network.java:
```diff
@@ -46,21 +48,22 @@ public void close() {
         try {
             socket = new Socket(SERVER_ADR, SERVER_PORT);
-            System.out.println("Сооединение с " + SERVER_ADR + ":" + SERVER_PORT + " установлено.");
+            LOGGER.info("Сооединение с " + SERVER_ADR + ":" + SERVER_PORT + " установлено.");
```
Так не совсем удачно форматируете строку для логирования. Надо:
`LOGGER.info("Сооединение с {}: {}  установлено.", SERVER_ADR, SERVER_PORT)`

Commit:
java3_homework7
Создать класс, который может выполнять «тесты».
В качестве тестов выступают классы с наборами методов, снабженных аннотациями @Test. Класс,
запускающий тесты, должен иметь статический метод start(Class testClass), которому в качестве
аргумента передается объект типа Class. Из «класса-теста» вначале должен быть запущен метод с
аннотацией @BeforeSuite, если он присутствует. Далее запускаются методы с аннотациями @Test, а
по завершении всех тестов – метод с аннотацией @AfterSuite.
К каждому тесту необходимо добавить приоритеты (int-числа от 1 до 10), в соответствии с которыми
будет выбираться порядок их выполнения. Если приоритет одинаковый, то порядок не имеет
значения. Методы с аннотациями @BeforeSuite и @AfterSuite должны присутствовать в
единственном экземпляре. Если это не так – необходимо бросить RuntimeException при запуске
«тестирования».
P.S. Это практическое задание – проект, который пишется «с нуля». Данная задача не связана
напрямую с темой тестирования через JUnit
